src/main/drivers/bus_spi.c

```diff
@@ -744,6 +744,19 @@ uint8_t spiGetExtDeviceCount(const extDevice_t *dev)
     return dev->bus->deviceCount;
 }
 
+// Link two segment lists
+// Note that there is no need to unlink segment lists as this is done automatically as they are processed
+void spiLinkSegments(const extDevice_t *dev, busSegment_t *firstSegment, busSegment_t *secondSegment)
```

Comment:
Maybe change argument order to `object, params...` This function is not (directly) working on device, but is appending something to segment chain (*firstSegment)

```suggestion
void spiLinkSegments(busSegment_t *firstSegment, const extDevice_t *dev, busSegment_t *secondSegment)
```

Reply:
I did it this way as the `dev` argument is always passed first in all other routines.

Reply:
IMO the difference is that other functions have `dev` as object they are working on. It becomes obvious when you try passing NULL to terminate segments.